includes/class-give-email-access.php

```diff
@@ -105,8 +105,8 @@ class Give_Email_Access {
 	 */
 	public function __construct() {
 
-		// get it started
-		add_action( 'init', array( $this, 'init' ) );
+		// Get it started.
+		add_action( 'wp', array( $this, 'init' ) );
```

Comment:
@mehul0810 This action hook must be processed only if current page history or success. Use already existing core functions like `give_is_success_page`